spinoso-string/src/lib.rs

```diff
@@ -1916,6 +1916,20 @@ impl String {
     /// length 1. For `String`s with [ASCII encoding] or [binary encoding],
     /// this function is equivalent to [`get`] with a range.
     ///
+    /// This type of ranges are allowed and they returns `""`
+    /// ```
+    /// [3.0.1] > "aaa"[1..0]
+    /// => ""
+    /// [3.0.1] > "aaa"[2..1]
+    /// => ""
+    /// ```
+    ///
+    /// but only if `start` is within the string.
+    /// ```
+    /// [3.0.1] > "aaa"[10..0]
+    /// => nil
+    /// ```
+    ///
```

Comment:
The comments were very helpful at least for me to understand the early return logic, but I was hesitant to duplicate them in all 3 encoded variants, so eventually I deleted most of them.

Reply:
thanks for adding them to the docs!